packages/dom/README.md

```diff
@@ -172,37 +168,29 @@ Given a DOM node, removes it from the DOM.
 
 _Parameters_
 
--   _node_ `Element`: Node to be removed.
-
-_Returns_
-
--   `void`: 
+-   _node_ `Node`: Node to be removed.
 
 <a name="replace" href="#replace">#</a> **replace**
 
 Given two DOM nodes, replaces the former with the latter in the DOM.
 
 _Parameters_
 
--   _processedNode_ `Element`: Node to be removed.
--   _newNode_ `Element`: Node to be inserted in its place.
-
-_Returns_
-
--   `void`: 
+-   _processedNode_ `Node`: Node to be removed.
+-   _newNode_ `Node`: Node to be inserted in its place.
 
 <a name="replaceTag" href="#replaceTag">#</a> **replaceTag**
 
 Replaces the given node with a new node with the given tag name.
 
 _Parameters_
 
--   _node_ `Element`: The node to replace
--   _tagName_ `string`: The new tag name.
+-   _node_ `Node`: The node to replace.
+-   _tagName_ `T`: The new tag name.
```

Comment:
Not sure the documentation generator is prepared for this.

Reply:
It looks like this is generated by [this part of the JSDoc](https://github.com/WordPress/gutenberg/pull/16352/files#r298657473) which is not to be found in the [JSDoc standard](https://jsdoc.app/).

Reply:
I'l defer this to @aduth but I was under the impression that we were not following the core jsdoc standard to the letter, but instead the one that is picked up by the language service (i.e. [tsdoc](https://github.com/microsoft/tsdoc))

Reply:
I think if we want to have any hope of achieving `tsc` types checking (#18838), we'll need to target the TypeScript syntax extensions. Even something such as sharing types across files (`import`) would not be straight-forward otherwise.

We'll need some clarity and guidelines around how this is rolled out, though. I've included it in the agenda for tomorrow's [#core-js meeting](https://make.wordpress.org/meetings/):

https://docs.google.com/document/d/1TJ06NiyLzThsplXdEhya51C13WE62nHN71kt997hXHw/edit#heading=h.quypt74go9w8

For example, some of the more advanced syntax like [intersection types](https://github.com/Microsoft/TypeScript/issues/16550) (often used for makeshift [type inheritence](https://github.com/Microsoft/TypeScript/issues/20077)) are both [explicitly unsupported in JSDoc](https://github.com/jsdoc/jsdoc/issues/1285) and flagged as invalid syntax by [`eslint-plugin-jsdoc`](https://npmjs.com/package/eslint-plugin-jsdoc) (more specifically, [`jsdoctypeparser`](https://www.npmjs.com/package/jsdoctypeparser), [related issue](https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50)). Some of this is rather open-ended on whether it _should_ be parsed correctly [[1]](https://github.com/jsdoc/jsdoc/issues/1710) [[2]](https://github.com/jsdoctypeparser/jsdoctypeparser/issues/96), but in the meantime we'll need to decide whether to use it, and how to work around potential interoperability issues.

Reply:
Thanks for the context of where this is coming from. Though, if we want to merge this particular PR before the changes that need to happen to our linting/docs tooling, we need to use the JSDoc syntax at the moment and revisit later.